react/AddToCartButton.tsx

```diff
@@ -28,6 +33,8 @@ interface Props {
   allSkuVariationsSelected: boolean
   text?: string
   unavailableText?: string
+  productLink: ProductLink
+  onClickBehavior: string
```

Comment:
Since this prop is actually an `enum`, it's always a good idea to enumerate the possible values in the type definition itself. That way, TypeScript can tell us exactly which values this could have. It would also break if you were to pass a `string` different from these two 😄 .
For this prop, it would be something like:

```suggestion
  onClickBehavior: 'add-to-cart' | 'go-to-product-page'
```